src/components/TextInput/styled.tsx

```diff
@@ -78,7 +78,7 @@ export const InputContainer = styled.div<Props>`
       `${theme.honeycomb.duration.normal} ease-in-out`,
     )};
 
-  ${({ isFocused }) => isFocused && focused};
+  ${({ isFocused, state }) => isFocused && state !== 'danger' && focused};
```

Comment:
I think you can just put line 74 (`${({ state, isPristine }) => state === 'danger' && !isPristine && danger};`) **after** `${({ isFocused }) => isFocused && focused};` to get the same effect in a simpler way.